lib/rack/bad_request.rb

```diff
@@ -0,0 +1,4 @@
+module Rack
+  module BadRequest
```

Comment:
If this is only used by the multipart parsing, it should be `Rack::Multipart::BadRequest` and moved to `rack/multipart.rb`

Reply:
I think the goal is to use it in other places, but this is the first place which makes the most sense. I could imagine potential usage in `Rack::Request`.

Reply:
I've added it to `query_parser.rb` too.